src/did/resolver.ts

```diff
@@ -1,29 +1,38 @@
-import { Resolver, DIDDocument } from "did-resolver";
+import { Resolver, DIDDocument, PublicKey } from "did-resolver";
```

Comment:
let's add a test to make sure the custom resolver works. One test will be enough, we can use msw + fake url to mock the response

Reply:
I tried to implement a mock result for a custom resolver, but it was very complicated because the resolver makes many JSON-RPC calls that are not easily mockable with a REST-dedicated tool like msw. 

After a few hours, I gave up. Not only because it's overcomplicated, but also because I realized that it didn't make sense. It's like testing the ethr-did-resolver library itself.

At this point, my conclusion is:

1. You have already tested all the verifiers with a resolver using Infura provider.
2. The goal of this PR is to allow passing another resolver.
3. Testing a custom resolver doesn't make sense, because "resolve" operations rely on external libs (did-resolver and ethr-did-resolver) and those libs are already tested.
4. Or, maybe I didn't understand what you expected me to test ;)

Reply:
I get what you mean, I will try on my side tomorrow and if I came to the same conclusion as your we will proceed as is

Reply:
Ok so I confirm there are internally at least 3 different calls to the URL provided. Let's not use msw.

We can however resolve a ropsten specific DID. The following DID `did:ethr:ropsten:0xf3beac30c498d9e26865f34fcaa57dbb935b0d74` will not resolve with the default resolver. We can create a test that provides a ropsten resolver. Here is an example by calling the resolver function directly:

```ts
    const resolver = createResolver({
      networks: [{ name: "ropsten", rpcUrl: `https://ropsten.infura.io/v3/${INFURA_API_KEY}` }],
    });
    const did = await resolver.resolve("did:ethr:ropsten:0xf3beac30c498d9e26865f34fcaa57dbb935b0d74");
```

Let's do it directly on the did verifier

Reply:
Ok, I successfully tested it using directly the resolve() function. But if I want to use the verifier instead (openAttestationDidIdentityProof), I need a signed document to verify. For that, I need the private key of did:ethr:ropsten:0xf3beac30c498d9e26865f34fcaa57dbb935b0d74, in order to sign the doc.

Reply:
Ok, so I created a new ropsten address to have a fresh new keypair. Then I tested as you proposed. It's ok, but there is a remaining problem : I can test successful scenario (with the custom resolver), but not failure scenario (without the custom resolver). Why ? Because when a did successfully resolved, it is added to a cache for 5 minutes. So, I cannot "expect" a failure when the custom resolver is not passed because the did document is retrieved from the cache. Of course, if I disable the cache by commenting that part of the code, it works. I don't think I should modify the resolver again to allow disabling the cache, so I'm going to push my tests with successful scenario only.

Reply:
https://github.com/Open-Attestation/oa-verify/pull/163/commits/03f487d8620b8f6b198405fb876001fb18e5cbf6

Reply:
I'm fine with the working path only for the moment. Thanks :)